src/main/kotlin/com/terraformation/backend/tracking/db/T0Store.kt

```diff
@@ -540,22 +457,168 @@ class T0Store(
+              .where(PLANTING_SITE_ID.eq(plantingSiteId))
+              .unionAll(
+                  DSL.select(
+                          ID.`as`("plot_id"),
+                          PLOT_T0_DENSITIES.SPECIES_ID.`as`("species_id"),
+                      )
+                      .from(this)
+                      .join(PLOT_T0_DENSITIES)
+                      .on(ID.eq(PLOT_T0_DENSITIES.MONITORING_PLOT_ID))
+                      .where(PLANTING_SITE_ID.eq(plantingSiteId))
               )
-              .limit(1)
-      )
+              .asTable("permanent_plot_species")
+
+      val permPlotIdField = permanentPlotSpecies.field("plot_id", ID.dataType)
+      val permSpeciesIdField = permanentPlotSpecies.field("species_id", SpeciesId::class.java)!!
+
+      val permPlotData =
+          dslContext
+              .select(
+                  ID,
+                  plantingSites.SURVIVAL_RATE_INCLUDES_TEMP_PLOTS,
+                  permSpeciesIdField,
+                  PLOT_T0_DENSITIES.PLOT_DENSITY,
+              )
+              .from(this)
+              .leftJoin(permanentPlotSpecies)
+              .on(ID.eq(permPlotIdField))
+              .leftJoin(PLOT_T0_DENSITIES)
+              .on(
+                  ID.eq(PLOT_T0_DENSITIES.MONITORING_PLOT_ID)
+                      .and(permSpeciesIdField.eq(PLOT_T0_DENSITIES.SPECIES_ID))
+              )
+              .leftJoin(PLANTING_SUBZONE_POPULATIONS)
+              .on(
+                  PLANTING_SUBZONE_ID.eq(PLANTING_SUBZONE_POPULATIONS.PLANTING_SUBZONE_ID)
+                      .and(permSpeciesIdField.eq(PLANTING_SUBZONE_POPULATIONS.SPECIES_ID))
+              )
+              .leftJoin(PLANTING_SUBZONES)
+              .on(PLANTING_SUBZONE_POPULATIONS.PLANTING_SUBZONE_ID.eq(PLANTING_SUBZONES.ID))
+              .where(PLANTING_SITE_ID.eq(plantingSiteId))
+              .and(IS_AD_HOC.eq(false))
+              .and(PERMANENT_INDEX.isNotNull)
+              .fetch { record ->
+                T0SitePermPlotData(
+                    monitoringPlotId = record[ID.asNonNullable()],
+                    survivalRateIncludesTempPlots =
+                        record[plantingSites.SURVIVAL_RATE_INCLUDES_TEMP_PLOTS.asNonNullable()],
+                    speciesId = record[permSpeciesIdField],
+                    plotDensity = record[PLOT_T0_DENSITIES.PLOT_DENSITY],
+                )
+              }
+
+      val isAllPermT0DataSet = permPlotData.all { it.plotDensity != null }
+
+      val survivalRateIncludesTempPlots =
+          if (permPlotData.isEmpty()) {
+            dslContext
+                .select(PLANTING_SITES.SURVIVAL_RATE_INCLUDES_TEMP_PLOTS)
+                .from(PLANTING_SITES)
+                .where(PLANTING_SITES.ID.eq(plantingSiteId))
+                .fetchOne(PLANTING_SITES.SURVIVAL_RATE_INCLUDES_TEMP_PLOTS.asNonNullable())!!
+          } else {
+            permPlotData.first().survivalRateIncludesTempPlots
+          }
```

Comment:
Disregard the following if this change is part one of a series and the things I'm talking about will make more sense in the context of later refactoring steps.

This feels to me like it could be simplified. The query above this is being used to answer two questions: is there any row with a null plot density, and do we need to fetch the survival-rate-includes-temp-plots flag separately?

For the first part of that, it seems like rather than constructing a list of objects, this could be a `SELECT EXISTS(SELECT 1 FROM ... WHERE plot_density IS NULL)`.

And for the second part, we can just fetch the flag from the planting site unconditionally; it's an extra database round-trip but the logic is simpler to follow than this "try to join if we can, but if that doesn't work, fetch it" logic.